torch/testing/_internal/distributed/rpc_utils.py

```diff
@@ -49,8 +52,38 @@
 # the agent, which then gets mixed-in with each test suite and each mp method.
 
 
+class RpcMultiProcessTestCase(MultiProcessTestCase):
+    def setUp(self):
+        super().setUp()
+        self.output_dir = tempfile.mkdtemp()
+        self.subprocess_init_data["output_dir"] = self.output_dir
+
+    def subprocess_init(self, data):
+        with open(os.path.join(data["output_dir"], f"rank_{self.rank}_stdout"), "wt") as f:
+            os.dup2(f.fileno(), 1)
+        sys.stdout = os.fdopen(1, "wt")
+        with open(os.path.join(data["output_dir"], f"rank_{self.rank}_stderr"), "wt") as f:
+            os.dup2(f.fileno(), 2)
+        sys.stderr = os.fdopen(2, "wt")
+
+    def on_test_failure(self):
+        for rank in range(self.world_size):
+            try:
+                with open(os.path.join(self.output_dir, f"rank_{rank}_stdout"), "rt") as f:
```

Comment:
Are these extra files cleaned up anywhere? After a test is over, they shouldn't stick around on the user's machine, right?